### ISSUE TITLE
[feature] add a storage layer to save diagrams

### ISSUE THREAD

**Issue by mingmxu** [plantuml/plantuml-server#330]
**Is your feature request related to a problem? Please describe.**

Currently PlantUML is a stateless service. There're challenges to share and find a diagram, as the URL is changed after editing, same for the image. It's also hard to resume the work after leaving for a while. 

**Describe the solution you'd like**

By adding a storage layer, plus a unique name for one diagram. It's convenient for multiple scenarios, like
* it has a static URL for one diagram, easy to share, bookmark;
* see all the diagrams in one place;
* pause/resume to edit a diagram without save/restore manually;

**Describe alternatives you've considered**

**Additional context**

**Comment by VladimirAlexiev** [comment]
@mingmxu https://real-world-plantuml.com/ has similar to what you asked for.

I think it goes a bit counter to the current spirit of the plantuml server to save diagrams... @The-Lum you are a "living lore" here, what do you think?

**Comment by mingmxu** [comment]
> I think it goes a bit counter to the current spirit of the plantuml server to save diagram

Right, it changes a lot as currently the server is supposed to be stateless. From user experience, we always need a place to save the diagrams, adding the feature to plantuml server is a straightforward option.
